pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1835,7 +1835,7 @@ protected void handleSend(CommandSend send, ByteBuf headersAndPayload) {
                 ? PositionImpl.get(send.getMessageId().getLedgerId(), send.getMessageId().getEntryId()) : null;
 
         // Persist the message
-        if (send.hasHighestSequenceId() && send.getSequenceId() <= send.getHighestSequenceId()) {
+        if (send.hasHighestSequenceId()) {
```

Comment:
It does not make sense without adding a test to show which issue it fixes.

IMO, the original code is just defensive programming that handles the exceptional case that `sequence_id` > `highest_sequence_id` when the client constructs the `CommandSend` request. Though with the correct implementation, the client should not set a sequence id that is greater than the highest sequence id.

Reply:
get it,thank you